### PR TITLE
Project 2 [feat]: Result inside form

### PR DESCRIPTION
**Pull request by mabackstr**
### Footprint calculation result is now included on the form page.

**Tasks: 10.1, 10.2**

When the submit to BE is performed, the BE no longer returns a redirect, but the footprint result.

When the fetch result is received, the "result" state is set to the resulting footprint object that was received.

When the result has been received, the form goes to the next step in the form, i.e. does not show the flight question but moves on to the result category.

The result is stored in SessionStorage, so that if the user goes back in the form but does not change any answer, the fetch call is not performed unnecessarily.

The category state has been removed due to its nature not fitting well with content that is not question-based. What is shown in the FootprintForm is now dependent on the "question"index (i.e. currentIndex).

In FootprintForm, JSX is now divided into two different pages: QuestionPage and ResultPage. While the user is in the calculator-based view of FootprintForm (answering questions), QuestionPage is used. When the user views result-related information, ResultPage is used. currentIndex (previously known as questionIndex) is now a state. With this, a state for the category is not needed and simplified the construction of the result page.

### CODE REVIEW

File: app/assets/main/react_components/footprint_form/FootprintForm.jsx [hunk 1]
```diff
@@ -119,21 +131,30 @@ const FootprintForm = ({ calculator, questionStrings, options, footprint, route
...
           'Content-Type': 'application/json',
         },
-        body: JSON.stringify(cleanFootprint)
+        body: JSON.stringify(answers)
       };
-      fetch(URL, requestOptions)
-      .then(res => window.location.href = res.url)
+    fetch(URL, requestOptions)
+      .then(response => {
+        response.json().then(calculatedFootprint => {
```
Comment (teolenn) [inline app/assets/main/react_components/footprint_form/FootprintForm.jsx:138]: Hur funkar det här?
Som jag förstår det är calculatedFootprint responsobjektet från servern, är det då det färdigberäknade objektet?

File: app/assets/main/react_components/footprint_form/FootprintForm.jsx [hunk 2]
```diff
@@ -1,44 +1,56 @@
...
+
+  function getSessionStorage(key){
+    const item = sessionStorage.getItem(key)
+    return item ? JSON.parse(item) : {}
+  }    
+
+  function areObjectsEqual(...objects){
+    return objects.every(obj => JSON.stringify(obj) === JSON.stringify(objects[0]));
+  }
+
```
Comment (teolenn) [inline app/assets/main/react_components/footprint_form/FootprintForm.jsx:53]: Kan överväga att ge dessa funktioner en liten kodkommentar även om de är korta

File: app/assets/main/react_components/footprint_form/QuestionPage.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+const QuestionPage = ({ question, options, onAnswerGiven, selectedKey, onNumericalInput, savedValue}) => {
+    return (
+        <div>
+            <Title text={question}/>
+            {!options.isNumerical ? 
+            <OptionList 
+                onAnswerGiven={(givenAnswer) => onAnswerGiven(givenAnswer)} 
+                options={options.options}
+                selectedKey={selectedKey} 
+            />
```
Comment (teolenn) [inline app/assets/main/react_components/footprint_form/QuestionPage.jsx:19]:
```suggestion
                <OptionList 
                    onAnswerGiven={(givenAnswer) => onAnswerGiven(givenAnswer)} 
                    options={options.options}
                    selectedKey={selectedKey} 
                />
```

File: app/assets/main/react_components/footprint_form/QuestionPage.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            :
+            <OptionNumerical 
+                onAnswerGiven={(givenAnswer) => onAnswerGiven(givenAnswer)} 
+                option={options} 
+                onNumericalInput={onNumericalInput}
+                savedValue={savedValue} 
+            />
```
Comment (teolenn) [inline app/assets/main/react_components/footprint_form/QuestionPage.jsx:26]:
```suggestion
                <OptionNumerical 
                    onAnswerGiven={(givenAnswer) => onAnswerGiven(givenAnswer)} 
                    option={options} 
                    onNumericalInput={onNumericalInput}
                    savedValue={savedValue} 
                />
```

**Review by teolenn — CHANGES_REQUESTED** [review]
Snyggt! Bara några pyttesmå grejer

File: app/assets/main/react_components/footprint_form/FootprintForm.jsx (hunk 2 above)
Comment (mabackstr) [reply app/assets/main/react_components/footprint_form/FootprintForm.jsx:53]: Hmm jag tänkte att namnen på funktionerna var nog beskrivande, men det kanske jag bör lägga till! @e-ry som tredje part får du säga vad du tycker!

**Review by teolenn — APPROVED** [review]

File: app/assets/main/react_components/footprint_form/Container.jsx [hunk 5]
```diff
@@ -2,22 +2,30 @@ import React from 'react';
...
-const Container = ({ footprint, calculator, lifestyleFootprintsText, slug }) => {
-
+const Container = ({ footprint, calculator, slug, lang, registrationsText, commonText, modelText, lifestyleFootprintsText }) => {
+  const texts = {
+    registrationsText: JSON.parse(registrationsText),
+    commonText: JSON.parse(commonText),
+    modelText: JSON.parse(modelText),
+    lifestyleFootprintsText: JSON.parse(lifestyleFootprintsText)
+  }
+
```
Comment (e-ry) [inline app/assets/main/react_components/footprint_form/Container.jsx:12]: Gillar den här lösningen

File: app/assets/main/react_components/footprint_form/QuestionPage.jsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+                    option={options} 
+                    onNumericalInput={onNumericalInput}
+                    savedValue={savedValue} 
+                />
+            }
+        </div>
+    )
+}
+
+export default QuestionPage
```
Comment (e-ry) [inline app/assets/main/react_components/footprint_form/QuestionPage.jsx:32]:
```suggestion
export default QuestionPage

```

Git tycker det ska va newline här, men bara i denna fil?? Mkt skumt

**Review by e-ry — APPROVED** [review]
Har inte kollat noga på allt, men jag tänker att när vi refaktorerar tillsammans kommer det sista lösa sig!

**Pull request merged**